components/CMakeLists.txt

```diff
@@ -91,9 +91,24 @@ endif()
 
 project(E3SM C CXX Fortran)
 
-# We do want CMAKE_BUILD_TYPE to be set, but we do NOT want CMake to 
+# Include function definitions
+include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/cmake_util.cmake)
+include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/build_mpas_model.cmake)
+include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/build_scream.cmake)
+include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/build_model.cmake)
+
+set(CMAKE_VERBOSE_MAKEFILE TRUE)
+
+if(USE_CUDA)
+  enable_language(CUDA)
+endif()
+
+# Scream manages its own flags
+build_scream()
+
+# We do want CMAKE_BUILD_TYPE to be set, but we do NOT want CMake to
```

Comment:
Gets me wondering: what's the point of setting the build type, if we do not want cmake to use it? Maybe, we could set `CMAKE_<LANG>_<BUILD_TYPE>` inside `cime_config/machines/cmake_macros/$mach.cmake`?

Reply:
@bartgol , we do want CMake to use it for SCREAM.

Reply:
True. I do think, however, that the whole e3sm should make use of the build type. The `$mach.cmake` files seem like an appropriate way to add default debug/release flags (mostly, optimization level and/or debug flag). Users should still be able to a) override optimization levels, and b) append additional flags.

Reply:
@bartgol I agree, there is a lot of legacy stuff built in to CIME that predates the cmake effort, so there's a lot of non standard stuff that still needs to be cmake-ified. This is on my longterm todo list.